radsat-sk/operation/crypt/encrypt.c

```diff
@@ -0,0 +1,58 @@
+#include "encrypt.h"
+
+int getPaddedBufSize(int plaintextLen);
+void padBuffer(uint8_t* buff, uint8_t* paddedBuff, int buffLen, int newBuffLen);
+void encrypt(uint8_t* plaintext, int plaintextLen);
+
+
+/**
+ * @brief Gets the next multiple of 16 from the size of the input buffer
+ * @header "comms-processing/crypt/encrypt.h"
+ * @param plaintextLen - The size of the unpadded buffer containing data to be encrypted.
+ * @pre need to have a variable storing the size of the plaintext buffer
+ * @return The size of the new buffer to be created
+ */
+int getPaddedBufSize(int plaintextLen){
+    // Special case for when the buffer size is already a multiple of 16:
+    if (plaintextLen % MODULUS == 0){
+	int newBufSize = plaintextLen + MODULUS;
```

Comment:
Why do we still have to pad even if its a multiple of 16?

Reply:
If we don't pad anything then the depadding function will look at the last byte and see some random garbage rather than the padding value and get confused.